src/writer.js

```diff
@@ -19,9 +19,20 @@ class Writer {
   }
 
   append (span) {
-    const trace = span.context()._trace
+    const spanContext = span.context()
+    const trace = spanContext._trace
+    const isCompletedTrace = (trace.started.length === trace.finished.length)
+
+    if (spanContext._sampling.drop === true) {
```

Comment:
This logic could be moved in the existing condition instead to avoid the double check.

Reply:
Yep totally makes sense, cleaned up in `1172d31`